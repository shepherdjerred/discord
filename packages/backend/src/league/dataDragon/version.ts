// @deno-types="npm:@types/lodash"
import _ from "npm:lodash@4.17.21";
import { z } from "https://esm.sh/zod@3.22.4";

const versions = z
  .array(z.string())
  .parse(
    await (
      await fetch("https://ddragon.leagueoflegends.com/api/versions.json")
    ).json()
  );

export const latestVersion = _.first(versions) as string;

if (latestVersion === undefined) {
  throw new Error("latest version is undefined");
}
