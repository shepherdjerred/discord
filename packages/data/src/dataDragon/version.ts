import _ from "lodash";
import { z } from "zod";

const versions = z
  .array(z.string())
  .parse(await (await fetch("https://ddragon.leagueoflegends.com/api/versions.json")).json());

export const latestVersion = _.first(versions) as string;

if (latestVersion === undefined) {
  throw new Error("latest version is undefined");
}
