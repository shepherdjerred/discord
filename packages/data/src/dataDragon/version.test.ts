import { describe, expect, test } from "vitest";
import { latestVersion } from "./version.js";

describe("dataDragon", () => {
  test("should be able to get version", () => {
    expect(latestVersion).toMatchSnapshot();
  });
});
