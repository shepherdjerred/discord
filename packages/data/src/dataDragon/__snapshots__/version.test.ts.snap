// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`dataDragon > should be able to get version 1`] = `"13.23.1"`;
